float64[] warning_levels
